corehq/apps/userreports/reports/view.py

```diff
@@ -255,8 +255,14 @@ def filter_values(self):
     @memoized
     def filter_context(self):
         return {
-            filter.css_id: filter.context(self.request_dict, self.request_user, self.lang)
-            for filter in self.filters
+            report_filter.css_id: report_filter.context(self.request_dict, self.request_user, self.lang)
+            for report_filter in self.filters
+        }
+
+    def filter_context_for_saved_report(self, saved_config):
```

Comment:
what would you think about moving this inside `filter_context`?  That way `self.filter_context` always returns the correct value, and is always applicable.  To do so, you'd probably need to pull the call that gets `saved_report_config` (lines 391 and 392) into a `memoized` or `@cached_property` so it can be referenced in both places.

Reply:
Yeah I like that. I made that change on the latest commit.